mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -1636,6 +1639,84 @@ private static ReadPreference getReadPreference(AssociationContext associationCo
 		return associationContext.getAssociationTypeContext().getOptionsContext().getUnique( ReadPreferenceOption.class );
 	}
 
+
+	@Override
+	public boolean supportsNamedPosition() {
+		return false;
+	}
+
+	/**
+	 * call stored procedure
+	 * @param storedProcedureName name of stored procedure.
+	 * @param params - array with values
+	 * @param tupleContext the tuple context
+	 *@see <a href="https://stackoverflow.com/questions/32480060/call-mongodb-function-from-java"> example</a>
+	 * @return
+	 */
+	@SuppressWarnings("unchecked")
+	@Override
+	public ClosableIterator<Tuple> callStoredProcedure(String storedProcedureName, Object[] params,
+			TupleContext tupleContext) {
+		StringBuffer commandLine = new StringBuffer( storedProcedureName ).append( "(" );
+		for ( Object param : params ) {
+			if ( param instanceof String ) {
+				commandLine.append( '\'' ).append( param.toString() ).append( '\'' );
+			}
+			else {
+				commandLine.append( param.toString() );
+			}
+			commandLine.append( "," );
+		}
+		commandLine.setLength( commandLine.length() - 1 );
+		commandLine.append( ")" );
+		Document result = provider.getDatabase().runCommand( new Document( "$eval", commandLine.toString() ) );
+		Double ok = result.getDouble( "ok" );
+		if ( ok.intValue() != 1 ) {
+			// return not ok! throw exception?
+		}
+		Object retvar = result.get( "retval" );
+		List<Tuple> resultTuples = null;
+		if ( retvar instanceof Document ) {
+			String firstRetValTag = ( (Document) retvar ).keySet().iterator().next();
+			Object firstRetVal = ( (Document) retvar ).get( firstRetValTag );
+			if ( firstRetVal instanceof List ) {
+				//it is collection of entities
+				List<Document> documents = (List<Document>) firstRetVal;
+				resultTuples = new ArrayList<>( documents.size() );
+				for ( Document doc : documents ) {
+					resultTuples.add( convert( doc ) );
+				}
+			}
+			else {
+				throw new NotYetImplementedException( "Not supported type " + firstRetVal.getClass() );
```

Comment:
What other value could this return?

Reply:
I don't know. 
Do you offer to remove the code?

Reply:
Could you create a test and check please?

Reply:
fixed.